core/domain/classifier_services.py

```diff
@@ -124,6 +124,35 @@ def classify_string_classifier_rule(state, normalized_answer):
 
     return None
 
+def check_classification_condition(state):
+    """Checks if conditions for training are satisfied.
+
+    Args:
+        state: Domain object for the State.
+
+    Returns:
+        bool: True, if the conditions are satisfied.
+    """
+    training_examples = [
```

Comment:
@seanlip , I think that would just be further breaking it down since that's almost what we are doing. If the interaction doesn't support classification, the training_examples list would be empty and the check would fail anyway.